sphinx/util/typing.py

```diff
@@ -90,17 +90,28 @@ def is_invalid_builtin_class(obj: Any) -> bool:
 Inventory = dict[str, dict[str, InventoryItem]]
 
 
-# return of a setup() function
-# https://www.sphinx-doc.org/en/master/extdev/index.html#extension-metadata
-class _ExtensionMetadata(TypedDict, total=False):
+class ExtensionMetadata(TypedDict, total=False):
+    """The metadata returned by an extension's ``setup()`` function.
+
+    See :ref:`ext-metadata`
+    """
+
     version: str
+    """a string that identifies the extension version (default ``'unknown version'``)."""
     env_version: int
+    """an integer that identifies the version of env data added by the extension."""
     parallel_read_safe: bool
+    """a boolean that specifies if parallel reading of source files
+    can be used when the extension is loaded  (default ``False``).
+    """
```

Comment:
```suggestion
    """Indicate whether parallel reading of source files is supported 
    by the extension.
    """
```

Technically, the default value is ``None`` since we ask extensions to make it explicit whether they are safe for parallel reading or not. There is a bit of inconsistency with `parallel_write_safe` which has a default value of ``True`` instead (so it will never be warned against technically, unless you explicitly set ``None`` as a value... but it's against the typed dict structure).